docs/reference/contributing/storage/BlockDevice.md

```diff
@@ -1,6 +1,10 @@
 <h2 id="contributing-block-device">Block Devices</h2>
 
-[Include a brief description here.]
+Block devices are the basic building block of storage solutions in Mbed OS. 
+
+![MbedOSStorage](https://s3-us-west-2.amazonaws.com/mbed-os-docs-images/MbedOS-storage-overview.png)
+  
+Filesystems are backed by blockdevice implementations. <--TODO: Link to said implementations? --> The BlockDevice API performs the low level interactions with the hardware storage. To add your own block device implementation it is recommended to inherit from the BlockDevice class. For details on how to extend the BlockDevice interface, please refer to the [class reference](#blockdevice-class-reference) and [Implementing BlockDevice](#implementing-blockdevice) sections below.
```

Comment:
Should I link to every implementation, a subset of them, or just to the storage page?

Reply:
I think the storage page should be sufficient. Good question